### PR TITLE
(PE-12001) Add solaris 10

### PR DESCRIPTION
**Pull request by jpartlow**
Based on #63.  This should not be merged until that PR is.

### CODE REVIEW

File: manifests/osfamily/solaris.pp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      class { puppet_agent::prepare::package:
+        package_file_name => $package_file_name,
+      }
+      contain puppet_agent::prepare::package
+
+      $_unzipped_package_name = regsubst($package_file_name, '\.gz$', '')
+      exec { "unzip ${package_file_name}":
+        path      => '/bin:/usr/bin:/sbin:/usr/sbin',
+        command   => "gzip -d /opt/puppetlabs/packages/${package_file_name}",
+        creates   => "/opt/puppetlabs/ppackages/${_unzipped_package_name}",
```
Comment (highb) [inline manifests/osfamily/solaris.pp]: Is `ppuppet` a typo? I assume this is fixed in your local changes.

Comment (jpartlow) [inline manifests/osfamily/solaris.pp]: That's a typo; I'll fix that, and rebase for a change to sles10 pr

**Comment by jpartlow** [comment]
Ah, right, I just figured out why the version constraints are on the osfamily classes; because the osfamily tests will see empty catalogs if the Puppet version is not 3.8.  I'll fix this up after arch.

**Comment by jpartlow** [comment]
Had to add two resources to stop the pe-puppet and pe-mcollective.  It's also necessary that they (or at least the pe-puppet service stop) happen at the very end or the run is interrupted.  There may be better ways to do this then what I currently have arranged.

**Comment by jpartlow** [comment]
Manually tested an upgrade of Solaris10 3.8.2 -> 2015.2.x with this patch.

**Comment by highb** [comment]
Verified upgrade from 3.8.2->2015.2.1, Sol10 agent from vmpool upgraded sucessfully, `mco ping` succeeds after upgrade and puppet runs complete.

**Pull request merged**